### DISCUSSION TITLE
Raw file

### DISCUSSION THREAD

**Discussion by JohnDoezNot** [dernasherbrezon/r2cloud#141] [Q&A]
What format are raw recorded files and how to convert them into a regular iq or wav file to open it in programs like sdrsharp? Thank you !

**Comment by dernasherbrezon** [comment]
Hi,

The format is described in the ```meta.json``` nearby. For example: ```"dataFormat": "COMPLEX_UNSIGNED_BYTE",```.

r2cloud supports [SigMF](https://github.com/sigmf/SigMF) specification. You can download files in that format from the UI:
<img width="1137" alt="Screenshot 2024-01-30 at 20 13 42" src="https://github.com/dernasherbrezon/r2cloud/assets/1614424/6798b317-a341-45c8-a5b2-7056f5612127">

If sdrsharp supports SigMF, then it will be able to load these files with no additional configuration.

**Reply by JohnDoezNot** [reply]
In my version it's only RAW file with samplerate 240000. It is playable it SDRSharp but audio is not decodable. I would like to know with what parameters (8 bits or 32 bits, 1 channel or 2 channels) should I open raw files in SDRSharp or in a converter like SOX

![изображение](https://github.com/dernasherbrezon/r2cloud/assets/165147633/d9efee59-c95c-4d3f-af21-475504ced8e8)

В моей версии есть только файл RAW с частотой дискретизации 240000. Он воспроизводится в SDRSharp, но сигнал (например, пакеты) не декодируются. Я хотел бы знать, с какими параметрами (8 бит или 32 бита, 1 канал или 2 канала) мне следует открывать raw файлы в SDRSharp или преобразовывать их в конвертере типа SOX. Я тоже русскоязычный, но пишу по-английски, вдруг кто-нибудь из англоязычных задастся таким же вопросом. А вообще приглашаю в телеграм-группу, посвященную радиолюбительским спутникам https://t.me/amateursat, так как о такой замечательной программе как Ваша даже большинство тех, кто в "теме" к сожалению, не слышал, хотя там есть и те, кто ей пользуется. Благодарю !

**Reply by dernasherbrezon** [reply]
Most likely it is IQ file 32-bit float. I.e. cf32 or complex float 32 bit.

**Reply by JohnDoezNot** [reply]
Thank you for your reply. But the signal becomes more or less similar to a packet when in SDRSharp I select 240,000 samples, 8 bits, 2 channels, although still the signals of even popular satellites are not decoded via the sound modem. Maybe the recording is done in 8 bits? And are there parameters in the system that determine the recording parameters? I mean some configuration file in Linux itself or some other settings.

**Reply by dernasherbrezon** [reply]
The signal format in the meta.json file which located near output.raw file. Check "dataFormat" value.  For example: "dataFormat": "COMPLEX_UNSIGNED_BYTE" means signal is unsigned 8bit, 2 channels.
